chain/jsonrpc-primitives/src/message.rs

```diff
@@ -130,6 +130,39 @@ impl std::string::ToString for RpcError {
     }
 }
 
+impl From<crate::blocks::RpcBlockError> for RpcError {
+    fn from(error: crate::blocks::RpcBlockError) -> Self {
+        let error_data = match error {
+            crate::blocks::RpcBlockError::BlockMissing(hash) => Some(Value::String(format!(
+                "Block Missing (unavailable on the node): {} \n Cause: Unknown",
+                hash.to_string()
+            ))),
+            crate::blocks::RpcBlockError::BlockNotFound(s) => {
+                Some(Value::String(format!("DB Not Found Error: {} \n Cause: Unknown", s)))
+            }
+            crate::blocks::RpcBlockError::Unexpected(s) => Some(Value::String(s)),
+            crate::blocks::RpcBlockError::NotSyncedYet
+            | crate::blocks::RpcBlockError::TooManyRequests(_) => {
+                Some(Value::String(error.to_string()))
+            }
+        };
+
+        Self::new(-32_000, "Server error".to_string(), error_data)
+    }
+}
+
+impl From<actix::MailboxError> for RpcError {
+    fn from(error: actix::MailboxError) -> Self {
+        Self::new(-32_000, "Server error".to_string(), Some(Value::String(error.to_string())))
+    }
+}
```

Comment:
Do we need it?

Reply:
For a while. Will be possible to delete after all refactoring.